cmake/TaichiCore.cmake

```diff
@@ -74,7 +74,20 @@ if (TI_WITH_CC)
   list(APPEND TAICHI_CORE_SOURCE ${TAICHI_CC_SOURCE})
 endif()
 
-add_library(${CORE_LIBRARY_NAME} SHARED ${TAICHI_CORE_SOURCE} ${PROJECT_SOURCES})
+# We don't mean to create a dedicated library just for common. Instead, we are
+# trying to identify, refactor and group the core files into a smaller lib. This
+# lib should be mostly self-contained and completely Taichi focused, free from
+# the "application" layer such as pybind11 or GUI.
+file(GLOB TAICHI_TESTABLE_SRC "taichi/common/*.cpp" "taichi/common/*.h")
+# TODO(#2196): Maybe we can do the following renaming in the end?
+# taichi_core --> taichi_pylib (this requires python-side refactoring...)
+# taichi_testable_lib --> taichi_core
+set(TAICHI_TESTABLE_LIB taichi_testable_lib)
```

Comment:
The goal of this entire PR is so that we can have this `taichi_testable_lib`, which could later be linked into the unit tests executable.